model/task/db.go

```diff
@@ -773,12 +773,12 @@ func FindNeedsContainerAllocation() ([]Task, error) {
 // needsContainerAllocation returns the query that filters for a task that
 // currently needs a container to be allocated to run it.
 func needsContainerAllocation() bson.M {
```

Comment:
It seems like this query is equivalent to the `task.FindAll` query in the check blocked tasks job, so it would be simpler to export it + use it instead of setting `ContainerAllocated = false` manually.

Reply:
It is distinct because the check blocked tasks job looks at [UndispatchedContainerTasksQuery](https://github.com/evergreen-ci/evergreen/pull/6900/files#diff-bb5e4c87753aeaf8606e661b6f8a86d27ed96e48981b45490846bec566ca54f8R111) which includes tasks that have yet to have their dependencies met, rather than `ScheduledContainerTasksQuery`

Reply:
Re-requesting review just to make sure this is resolved